src/main.rs

```diff
@@ -30,35 +29,47 @@ fn main() -> Result<()> {
         let entry = entry.unwrap();
         let f_name = entry.file_name().to_string_lossy();
         let path = entry.path();
-        match std::panic::catch_unwind(|| {
-            //if !path.to_string_lossy().contains("nixpkgs/doc/default.nix") { return; }
+        //match std::panic::catch_unwind(|| {
+        //if !path.to_string_lossy().contains("nixpkgs/doc/default.nix") { return; }
 
-            if f_name.ends_with(".nix") {
-                //println!("{}", path.display());
+        if f_name.ends_with(".nix") {
+            //println!("{}", path.display());
 
-                // check whether this here is cache-wise better or if reading in chunks is better
-                let file = fs::read(path).unwrap();
+            // ./target/release/nixy | sort -n
 
-                let lexer = NixLexer::new(&file).filter(|t| match t.token_type {
-                    NixTokenType::Whitespace(_)
-                    | NixTokenType::SingleLineComment(_)
-                    | NixTokenType::MultiLineComment(_) => false,
-                    _ => true,
-                });
+            // check whether this here is cache-wise better or if reading in chunks is better
+            // in chunks should be better, haskell is 11MB
 
-                for token in lexer.clone() {
-                    //println!("{:?}", token.token_type);
-                }
+            // TODO FIXME read block by block
+            let file = fs::read(path).unwrap();
+            println!("{} {}", file.len(), path.display());
 
-                parse(&mut multipeek(lexer));
+            let lexer = NixLexer::new(&file).filter(|t| match t.token_type {
+                NixTokenType::Whitespace(_)
+                | NixTokenType::SingleLineComment(_)
+                | NixTokenType::MultiLineComment(_) => false,
+                _ => true,
+            });
```

Comment:
I just discovered the [`matches!`](https://doc.rust-lang.org/std/macro.matches.html#examples) macro, it is very nice for these cases

Reply:
Thanks, I'm aware because clippy also knows it. Currently this is a write ugly code fast because my primary goal is to find out if all of this is worth it. If it probably is I will clean up things.